sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -125,6 +125,12 @@ private[hive] object HiveShim {
     // for Serialization
     def this() = this(null)
 
+    override def equals(other: Any): Boolean = other match {
+      case a: HiveFunctionWrapper =>
+        functionClassName == a.functionClassName
```

Comment:
@yhuai Have addressed your comments. Can you please take a look to see if it looks ok ?